### PR TITLE
Frontend Homework 8

### PR DESCRIPTION
**Pull request by asish918**
## Drift Video
[Drift](https://video.drift.com/v/abWpMNQZpSQn3S3sl9BZDPU67HRKJWtqHiX6dsStvmCw)

## Screenshots

All Recipes
SCR-20240215-ugzq

Search Modal Result
Screenshot 2024-02-15 at 23 31 35

### CODE REVIEW

**Comment by asish918** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 0 |
| code_smells | 4 |
| reliability_rating | A (0 Bugs)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

**Comment by Nitesh2503** [comment]
- Proper commit message's format must be used
- Follow the proper TS directory structure
- Avoid DOM manipulation

File: frontend/homework/8/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+interface AppRecipe {
+    image: string,
+    name: string,
+    rating: number,
+    cuisine: string,
+    ingredients: string[],
+    difficulty: string,
+    timeTaken: number,
+    calorieCount: number,
+}
```
Comment (Nitesh2503) [inline frontend/homework/8/src/app.ts:36]: Use different file for this

**Comment by Nitesh2503** [comment]
Work on modularity, code readability

**Pull request closed without merging**